lib/Sema/CodeSynthesisDistributedActor.cpp

```diff
@@ -81,9 +96,519 @@ static VarDecl *addImplicitDistributedActorIDProperty(
+
+  auto body = BraceStmt::create(C, sloc, {ifStmt}, sloc, implicit);
+  return {body, /*isTypeChecked=*/false};
+}
+
+static FuncDecl *createDistributedThunkFunction(FuncDecl *func) {
+  auto &C = func->getASTContext();
+  auto DC = func->getDeclContext();
+
+  auto systemTy = getConcreteReplacementForProtocolActorSystemType(func);
+  assert(systemTy &&
+         "Thunk synthesis must have concrete actor system type available");
+
+  DeclName thunkName = createDistributedFuncName(C, func);
+
+  // --- Prepare generic parameters
+  GenericParamList *genericParamList = nullptr;
+  if (auto genericParams = func->getGenericParams()) {
+    genericParamList = genericParams->clone(DC);
+  }
+
+  GenericSignature thunkGenSig =
+      buildGenericSignature(C, func->getGenericSignature(),
+                            /*addedParameters=*/{},
+                            /*addedRequirements=*/{});
+
+  // --- Prepare parameters
+  auto funcParams = func->getParameters();
+  SmallVector<ParamDecl*, 2> paramDecls;
+  for (unsigned i : indices(*func->getParameters())) {
+    auto funcParam = funcParams->get(i);
+
+    auto paramName = funcParam->getParameterName();
+    // If internal name is empty it could only mean either
+    // `_:` or `x _: ...`, so let's auto-generate a name
+    // to be used in the body of a thunk.
+    if (paramName.empty()) {
+      paramName = C.getIdentifier("p" + llvm::utostr(i));
+    }
+
+    auto paramDecl = new (C)
+        ParamDecl(SourceLoc(),
+                  /*argumentNameLoc=*/SourceLoc(), funcParam->getArgumentName(),
+                  /*parameterNameLoc=*/SourceLoc(), paramName, DC);
+
+    paramDecl->setImplicit(true);
+    paramDecl->setSpecifier(funcParam->getSpecifier());
+    paramDecl->setInterfaceType(funcParam->getInterfaceType());
+
+    paramDecls.push_back(paramDecl);
+  }
+  ParameterList *params = ParameterList::create(C, paramDecls); // = funcParams->clone(C);
+
+    auto thunk = FuncDecl::createImplicit(C, swift::StaticSpellingKind::None,
+                           thunkName, SourceLoc(),
+                           /*async=*/true, /*throws=*/true,
+                           genericParamList,
+                           params,
+                           func->getResultInterfaceType(),
+                           DC);
+  thunk->setSynthesized(true);
+  thunk->getAttrs().add(new (C) NonisolatedAttr(/*implicit=*/true));
+  thunk->setGenericSignature(thunkGenSig);
+  thunk->copyFormalAccessFrom(func, /*sourceIsParentContext=*/false);
+  thunk->setBodySynthesizer(deriveBodyDistributed_thunk, func);
+
+  return thunk;
+}
+
+/******************************************************************************/
+/*********************** SYNTHESIS ENTRY POINTS *******************************/
+/******************************************************************************/
+
+FuncDecl *GetDistributedThunkRequest::evaluate(
+    Evaluator &evaluator, AbstractFunctionDecl *afd) const {
```

Comment:
Nit: A more descriptive name would be `thunk` instead of `afd`.

Reply:
This is the "func" for which we make the "thunk" so I'll call it func I guess :)